base/server/src/com/netscape/cms/profile/def/SubjectKeyIdentifierExtDefault.java

```diff
@@ -197,20 +221,69 @@ public String getText(Locale locale) {
      */
     public void populate(IRequest request, X509CertInfo info)
             throws EProfileException {
-        SubjectKeyIdentifierExtension ext = createExtension(info);
+
+        // See if we have a SKI extenrion in the request already
+
+        CertificateExtensions extensions = null;
+
+        extensions = request.getExtDataInCertExts(EnrollProfile.REQUEST_EXTENSIONS);
+
+        SubjectKeyIdentifierExtension ext = createExtension(info, extensions);
 
         addExtension(PKIXExtensions.SubjectKey_Id.toString(), ext, info);
     }
 
-    public SubjectKeyIdentifierExtension createExtension(X509CertInfo info) {
-        KeyIdentifier kid = getKeyIdentifier(info);
+    public SubjectKeyIdentifierExtension createExtension(X509CertInfo info, CertificateExtensions extensions) {
+
+        //Check to see if we care if the cert request contains a SKI extension
+        boolean useSKIFromRequest = Boolean.valueOf(getConfig(CONFIG_USE_SKI_IF_IN_REQUEST,"false")).booleanValue();
+        org.mozilla.jss.netscape.security.x509.SubjectKeyIdentifierExtension keyIdExt = null;
+        if (extensions != null && useSKIFromRequest) {
+            if (extensions != null) {
+                Enumeration<Extension> e = extensions.getAttributes();
+
+                while (e.hasMoreElements()) {
+                    org.mozilla.jss.netscape.security.x509.Extension ext = e.nextElement();
+                    if (ext.getExtensionId().toString().equals(PKIXExtensions.SubjectKey_Id.toString())) {
+                        keyIdExt = (SubjectKeyIdentifierExtension) ext;
+                        logger.debug("SubjectKeyIdentifierExtDefault: found SubjectKeyIdentifier ext in request.");
+                        break;
+                    }
+                }
```

Comment:
Couldn't you just do this oneline?
keyIdExt = (Extension) extensions.get(PKIXExtensions.SubjectKey_Id.toString());